packages/analytics/client/src/events/types.ts

```diff
@@ -98,8 +98,9 @@ export enum TelemetryCounterType {
 export interface TelemetryCounter {
   /**
    * Indicates if the event contains data about succeeded, failed or dropped events.
+   * Refer to {@link TelemetryCounterType} for more details about each possible value.
    */
-  type: TelemetryCounterType;
+  type: keyof typeof TelemetryCounterType;
```

Comment:
list the potential values from the enum so users can set the values if they want to without importing the `enum`. This reduces bundle size because there is no need to import the `@kbn/analytics-client` runtime code only for this.

Reply:
I chose this approach mostly for documentation purposes. The `type` format is harder to have documentation for each potential value, while `enum` seems more straightforward.

But happy to remove the `enum`. I bet it will also save some bundle size 😇

Reply:
Removed in [`42d6bd0` (#135061)](https://github.com/elastic/kibana/pull/135061/commits/42d6bd005dea6c24cabab8533af7d8bfb76992ed)